ls1intum/Artemis
# Feature/Programming Exercise/Activate tutor dashboards

### Checklist

### Motivation and Context



Enable the tutor dashboards for programming exercises as discussed in https://github.com/ls1intum/Artemis/issues/940.

:information_source: ~Currently blocked by~ https://github.com/ls1intum/Artemis/pull/965.

### Description


**Supported with this PR:**
- Show programming exercises with assessment type manual on the leaderboards
- Confirm that the instructions were read and understood
- Get next assessment for a programming exercise / show assessment list
- Calculate scores / stats (including tutor leaderboard)
- Partial double blind (student info hidden in REST responses and UI)

**Not included in this PR:**
- Complaints / more feedback
- Lock mechanism
- Assessment within Artemis (see https://github.com/ls1intum/Artemis/issues/932)
- Complete double blind (traces of the student are part of the repository, e.g. in the zip folder name or the mvn project)

**Todos:**
- [x] Enable the course dashboard
- [x] Enable the exercise dashboard

**Bugs:**
- [x] Statistics are incorrect for multiple submissions per student (should only count 1 submission per student)
- [x] Remove student from next submission => should work already, test with tutor later
- [x] After manually assessing all submissions, a submission with a manual result is presented as the next one (probably happens because `loadAll()` is called on refresh)
- [x] Refresh page after result create/update
- [x] Result shows 'new result' even though has manual result
- [x] Fix circular serialization issue on update

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a programming exercise with a due date and the manual assessment enabled
4. Go to course administration > tutor dashboard
5. Open programming exercise in tutor dashboard
6. Assess submissions, check results, ...

### Screenshots



![tutor-dashboard](https://user-images.githubusercontent.com/28230611/68130198-37fc6b00-ff1b-11e9-8892-7d9888090589.gif)

(Note: The translation bug in the modal is fixed)
